### PR TITLE
Add FCC sugar for specifying kernel arguments

### PR DESCRIPTION
**Pull request by sohankunkerkar**
This is a rough draft to add FCC sugar for specifying kargs. A few changes are expected  based on how [this](https://github.com/coreos/ignition/pull/1178) PR evolves over a period time.

### CODE REVIEW

File: base/v0_4_exp/schema.go [hunk 1]
```diff
@@ -102,6 +103,15 @@ type IgnitionConfig struct {
 	Replace Resource   `yaml:"replace"`
 }
 
+type KernelArgument []KernelArgumentItem
+
+type KernelArgumentItem string
+
+type KernelArguments struct {
+	ShouldExist    KernelArgument `yaml:"should_exist"`
+	ShouldNotExist KernelArgument `yaml:"shouldnot_exist"`
```
Comment (bgilbert) [inline base/v0_4_exp/schema.go:112]: `should_not_exist`

**Review by bgilbert — COMMENTED** [review]
LGTM generally.

**Comment by bgilbert** [comment]
This should eventually be rebased on https://github.com/coreos/fcct/pull/223 and the `openshift/4.9.0-experimental` spec docs updated to match `fcos/1.4.0-experimental`.

**Comment by arithx** [comment]
Current code looks sane to me. What does everyone think about having the MachineConfig top-level `KernelArguments` also propogated to the Ignition `kernelArguments.shouldExist` field?

**Comment by bgilbert** [comment]
If we did that, kargs specified via FCCT would be applied to the system via a different mechanism than those specified by writing a MachineConfig directly.  Ideally we'd select one flow and document that everyone should use it.  Let's land the PR as-is for now, but create a JIRA card for 4.9 to select and implement a single recommended flow.

**Comment by arithx** [comment]
> If we did that, kargs specified via FCCT would be applied to the system via a different mechanism than those specified by writing a MachineConfig directly. Ideally we'd select one flow and document that everyone should use it. Let's land the PR as-is for now, but create a JIRA card for 4.9 to select and implement a single recommended flow.

Ah yeah, that's a fair point.

**Comment by sohankunkerkar** [comment]
Since https://github.com/coreos/ignition/pull/1178 has already merged, marking this PR ready for review

**Review by arithx — APPROVED** [review]

File: docs/config-fcos-v1_4-exp.md [hunk 2]
```diff
@@ -201,6 +201,9 @@ The Fedora CoreOS configuration is a YAML document conforming to the following s
     * **_threshold_** (int): sets the minimum number of pieces required to decrypt the device.
   * **_mirror_** (object): describes mirroring of the boot disk for fault tolerance.
     * **_devices_** (list of strings): the list of whole-disk devices (not partitions) to include in the disk array, referenced by their absolute path. At least two devices must be specified.
+* **_kernel_arguments_** (object): describes the desired kernel arguments.
```
Comment (bgilbert) [inline docs/config-fcos-v1_4-exp.md:204]: Nit: this is Ignition syntax, so let's put it before the sections that are Butane sugar (`boot_device`, and in `openshift` specs, `openshift`).

**Review by bgilbert — APPROVED** [review]

**Comment by bgilbert** [comment]
Followup issue in https://github.com/coreos/butane/issues/237.

File: docs/config-fcos-v1_4-exp.md (hunk 2 above)
Comment (bgilbert) [reply docs/config-fcos-v1_4-exp.md:204]: Yup.

**Review by bgilbert — APPROVED** [review]

**Pull request merged**